src/libraries/System.Net.Sockets/tests/FunctionalTests/CreateSocketTests.cs

```diff
@@ -445,6 +455,194 @@ public async Task Ctor_SafeHandle_Listening_Success(bool shareSafeHandle)
             }
         }
 
+        [DllImport("libc")]
+        private static extern int socket(int domain, int type, int protocol);
+
+        private const int PF_NETLINK = 16;
+
+        private class NlEndPoint : EndPoint
+        {
+            [StructLayout(LayoutKind.Sequential)]
+            internal struct sockaddr_nl
+            {
+                internal ushort sin_family;
+                private ushort pad;
+                internal int pid;
+                private int nl_groups;
+            }
+
+            private readonly int _pid;
+
+            public NlEndPoint(int pid)
+            {
+                _pid = pid;
+            }
+
+            public override AddressFamily AddressFamily
+            {
+                get
+                {
+                    return AddressFamily.Unknown;
+                }
+            }
+
+            public class NlSocketAddress : SocketAddress
+            {
+                // We need to create base from something known.
+                public unsafe NlSocketAddress(int pid) : base(AddressFamily.Packet)
+                {
+                    Span<sockaddr_nl> addr = stackalloc sockaddr_nl[1];
```

Comment:
You shouldn't need a span for this. You can just declare an instance and take its address.